apps/cave/contexts/Wagmi/WagmiContext.tsx

```diff
@@ -1,42 +1,46 @@
 import { SafeConnector } from '@gnosis.pm/safe-apps-wagmi'
-import { NEXT_PUBLIC_ALCHEMY_ID, NEXT_PUBLIC_INFURA_ID } from 'lib/env.conf'
+import { NEXT_PUBLIC_ALCHEMY_ID, NEXT_PUBLIC_INFURA_ID, NEXT_PUBLIC_LOCALHOST_RPC } from 'lib/env.conf'
 import { ReactNode, useEffect } from 'react'
 import { chain, Connector, createClient, defaultChains, useConnect, WagmiConfig } from 'wagmi'
 import { CoinbaseWalletConnector } from 'wagmi/connectors/coinbaseWallet'
 import { InjectedConnector } from 'wagmi/connectors/injected'
 import { MetaMaskConnector } from 'wagmi/connectors/metaMask'
 import { WalletConnectConnector } from 'wagmi/connectors/walletConnect'
-import { concaveProvider, concaveRPC, concaveWSProvider } from '../../lib/providers'
+import { concaveProvider, concaveProviderConfig, concaveWSProvider, hardhatProviderConfig } from '../../lib/providers'
+import { ImpersonateConnector } from './Connectors/ImpersonateConnector'
 import { UnstoppableConnector } from './Connectors/UnstoppableConnector'
 
-const chains = [chain.mainnet, chain.goerli] // app supported chains
-export const supportedChainsId = chains.map((c) => c.id)
+export const chains = [chain.localhost, chain.goerli, chain.mainnet] // app supported chains
 
+export const supportedChainsId = chains.map((c) => c.id)
 const connectors = [
   new SafeConnector({ chains }),
   new InjectedConnector({ chains }),
   new MetaMaskConnector({ chains }),
-  new WalletConnectConnector({ chains, options: { 
-    qrcode: false,
-    infuraId: NEXT_PUBLIC_INFURA_ID,
-    rpc: {
-      1: `https://eth-mainnet.alchemyapi.io/v2/${NEXT_PUBLIC_ALCHEMY_ID}`,
-    },
- } }),
+  new WalletConnectConnector({
+    chains, options: {
+      qrcode: false,
+      rpc: {
+        [chain.mainnet.id]: concaveProviderConfig.rpc,
+        [chain.localhost.id]: hardhatProviderConfig.rpc,
+      },
+    }
+  }),
   new UnstoppableConnector({ chains }),
   new CoinbaseWalletConnector({
     chains,
     options: {
       appName: 'Concave App',
-      jsonRpcUrl: concaveRPC,
+      jsonRpcUrl: concaveProviderConfig.rpc,
       appLogoUrl: 'https://app.concave.lol/assets/tokens/cnv.svg',
       darkMode: true,
       headlessMode: true,
     },
   }),
-] as Connector[]
+  NEXT_PUBLIC_LOCALHOST_RPC ? new ImpersonateConnector({ rpc: hardhatProviderConfig.rpc }) : undefined,
```

Comment:
@greg-schrammel do you have another ideia about it?

Reply:
maybe in package.json scripts?
something like `dev: next dev` `dev-impersonate: IMPERSONATE=true next dev && hardhat dev`
not sure